src/snowflake/connector/network.py

```diff
@@ -1070,28 +1106,17 @@ def _use_requests_session(self):
                 session.close()
         else:
             try:
-                session = self._idle_sessions.pop()
-            except IndexError:
-                session = self.make_requests_session()
-            self._active_sessions.add(session)
-            logger.debug(
-                "Active requests sessions: %s, idle: %s",
-                len(self._active_sessions),
-                len(self._idle_sessions),
-            )
+                hostname = urlparse(url).hostname
+            except Exception:
+                hostname = None
+
+            session_pool: SessionPool = self._sessions_map[hostname]
+            session = session_pool.get_session()
+            logger.debug(f"Session status for SessionPool '{hostname}', {session_pool}")
             try:
                 yield session
             finally:
-                self._idle_sessions.appendleft(session)
-                try:
-                    self._active_sessions.remove(session)
-                except KeyError:
-                    logger.debug(
-                        "session doesn't exist in the active session pool. "
-                        "Ignored..."
-                    )
+                session_pool.return_session(session)
                 logger.debug(
-                    "Active requests sessions: %s, idle: %s",
-                    len(self._active_sessions),
-                    len(self._idle_sessions),
+                    f"Session status for SessionPool '{hostname}', {session_pool}"
```

Comment:
I'm not sure whether this has the effect you wanted, `session_pool` should be casted to string here. You only defined `__repr__` and not `__str__`. Please verify that this logs what you expect, I might also be wrong and maybe Python tries `__str__` and then `__repr__`.

Reply:
The logging worked but you're right, it looks like `__str__` is used when its available, I'll change this to be `__str__`